evm_arithmetization/src/cpu/kernel/asm/core/terminate.asm

```diff
@@ -212,6 +212,8 @@ global terminate_common:
 
     // Go back to the parent context.
     %mload_context_metadata(@CTX_METADATA_PARENT_CONTEXT)
+    // Set the pruning flag.
+    PUSH 1 ADD
     SET_CONTEXT
```

Comment:
Maybe add some macro like `set_and_prune_context` for doing this? You can save repeating the comment "set the pruning flag"